src/HtmlReader/architecture.md

```diff
@@ -0,0 +1,78 @@
+- `LoadingIframeState`: We have rendered the iframe once, and are now waiting for it to fire off it's own `window.onLoad` function indicating that it has loaded all the html in the resource string we passed it.
+- `NavigatingState`: The iframe has finished loading, and now we are waiting for our `useEffect` hook to complete any inter-resource navigation that might be necessary depending on the current `state.location`. For example, scrolling the user to a specific page.
+- `ReadyState`: We have now navigated the user to a specific page, and are now finished with our work.
+
+The definitions for these states can be found in `./types.ts`. Note that state transitions should happen in the order listed above, with the exception that it is possible to go backwards occasionally. For example, from the `ReadyState`, we may go back to `FetchingResourceState` if the user switches to a new resource, or we might simply go back to the `NavigatingState` if they navigate to a new location in the same resource.
+
+We provide the text value `state.state` as a string describing the state. This makes it very easy to perform type narrowing. If `state.state === 'READY'`, typescript knows that `state.iframe` and `state.resource`, and more all exist.
+
+### Reducer
+
+It is recommended to look through the `./reducer` file to see how various actions (found defined in `./types.ts`) update the sate.
+
+## Fetching resources
+
+When the state is `FetchingResourceState`, we need to actually go out and fetch the resource referenced in `state.location.href`. We do this in the `useResource` hook. When we transition to a `FetchingResourceState`, that hook will call `getContent` to get the resource. It then will dispatch a `RESOURCE_FETCH_SUCCESS` action, which will transition the state to `RenderingIframeState`.
+
+### Updating the resource
+
+After fetching the resource, there are a couple things we need to do to it before we can render it:
+
+- Set a `<base>` element in the `<head>` with the url of the resource. This tells the iframe where relative urls within the resource are relative to.
+- Inject any injectables. For example, we need to add the Readium-CSS stylesheets to the `<head>`.
+- Set initial CSS variables like scroll settings and color mode.
+- Add a simple JS script that listens for internal link clicks and calls `window.parent.postMessage` to send a message to the parent about the clicked link.
+
+Note that we _could_ do all of this in a `useEffect` hook after the resource loads, but then there would be a flash of unstyled content between the first render and when we add our Readium-CSS stylesheet. To avoid this, we simple add it to the document string before it ever gets to the `<iframe>` for rendering.
```

Comment:
> we simple add it to the document string
nit: typo, but also:  what does the "it" here refer to?  The Readium-css stylesheet?

Reply:
Yea, I meant "we simply add all injectables to the document before it gets to the iframe".

Reply:
OK so it goes document -> put injectibles into document -> document + injectibles get to the iFrame.  And that's we make sure things are styled on first render.  Got it.

Reply:
Exactly